packages/functional-tests/src/runner/test-runner.spec.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import 'reflect-metadata';
+
+import { Logger, LogLevel } from 'logger';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { TestEnvironment } from '../common-types';
+import { test } from '../test-decorator';
+import { TestRunner } from './test-runner';
+
+// tslint:disable: no-unsafe-any
+
+class TestGroupStub {
+    @test(TestEnvironment.insider)
+    public async testA(): Promise<void> {
+        throw new Error('Error while invoked test A');
+    }
+
+    public testB(): void {
+        console.log('Invoked test B');
+    }
+
+    @test(TestEnvironment.all)
+    public testC(): void {
+        console.log('Invoked test C');
+    }
+
+    @test(TestEnvironment.canary)
```

Comment:
We can do `@test(TestEnvironment.all | TestEnvironment.all)` right? That's cool.

Reply:
right